projects/packages/videopress/src/class-initializer.php

```diff
@@ -348,6 +348,33 @@ public static function register_videopress_video_block() {
 			return;
 		}
 
+		// check current theme
+		$is_block_theme = wp_get_theme()->is_block_theme();
+
+		// for non block themes frontend, we defer the enqueuing to the frontend, so we're able to tell if we need the assets
+		if ( ! $is_block_theme && ! is_admin() ) {
+			add_action(
+				'wp_enqueue_scripts',
+				function () use ( $videopress_video_metadata_file ) {
+					if ( ! has_block( 'videopress/video', get_the_content() ) && ! has_block( 'jetpack-videopress/video', get_the_content() ) ) {
```

Comment:
If I'm correct, this condition only works in the post-content context. What happens if there is a VideoBlock video block instance in the footer added by a widget?

The second condition caught my attention because I've never seen the `jetpack-videopress` namespace before. Where is it being used?

Reply:
Good questions!

First one, most likely you're correct. Any ideas how to run the same parsing/check throughout all template parts besides the post content?

Second one, my bad, I did a search and misunderstood the output, I thought there was a jetpack-videopress/video block, I'll remove it

Reply:
It's like [has_block()](https://developer.wordpress.org/reference/functions/has_block/) accepts a string as a secondary parameter. Can we get the site content instead of the post one? That seems to be kinda tricky. I Never did it, honestly.

Reply:
> It's like [has_block()](https://developer.wordpress.org/reference/functions/has_block/) accepts a string as a secondary parameter. Can we get the site content instead of the post one? That seems to be kinda tricky. I Never did it, honestly.

I just remembered this issue only happens with  _classic_ themes, block themes render just fine and don't load the CSS unnecessarily. Moreover, block themes can use a VP block on templates and that also works properly.

Just for research purposes, I've been asking about some filter or global function to return the site (although we're talking about the page render, right?), just like `get_the_content()` does for posts, and it seems there is no such equivalent. I've been advised to maybe move to a more atomic filter (like [render_block](https://github.com/WordPress/gutenberg/issues/38120)) and load any assets there. I don't think we need this right now, the block is working fine in that regard.